src/app/providers/compression/compression.ts

```diff
@@ -11,11 +11,14 @@ export class CompressionProvider {
     const unzippedJson = gunzipSync(gzippedBytes).toString();
     return JSON.parse(unzippedJson);
   }
-
   extractTestSlotResult(compressedData: string): TestSlot {
     const gzippedBytes = Buffer.from(compressedData, 'base64');
     const unzippedJson = gunzipSync(gzippedBytes).toString();
     return JSON.parse(unzippedJson) as TestSlot;
   }
-
+  extractUnformatted(compressedData: string) {
+    const gzippedBytes = Buffer.from(compressedData, 'base64');
+    const unzippedJson = gunzipSync(gzippedBytes).toString();
+    return JSON.parse(unzippedJson);
+  }
```

Comment:
Ok so this is a direct copy of `extractTestResult` with the exception of the returned type. So instead of that what we can do is have a generic `extract` method and we can use Type generics like so:

```
  extract<T>(compressedData: string): T {
    const gzippedBytes = Buffer.from(compressedData, 'base64');
    const unzippedJson = gunzipSync(gzippedBytes).toString();
    return JSON.parse(unzippedJson);
  }
```

When we call the extract method then, we can do it like so

`this.compressionProvider.extract<TestResultSchemasUnion>(data)` 
or
`this.compressionProvider.extract<Object>(data)`